### PR TITLE
[PF-1817] Add lastUpdatedDate to workspace description

### PR DESCRIPTION
**Pull request by yuhuyoyo**
i'm piggyback on which is previously `WorkspaceAndHighestRole` and added `lastUpdatedDate` there. I'm planning to add CreatedDate in `WorkspaceAndAdditionalAttributes` as well in a following pr.

I'm putting the `WorkspaecAndAdditionalattributes` construction logic in workspaceService instead of workspaceapicontroller to force caller to get those instead of `workspace` and include these additional attributes in the workspace description. 

The other way is to have the construction logic in the controller but because we don't have unit test for the controller, testing it there is rather awkward and I don't like the controller to do the complicated thing.

### CODE REVIEW

File: openapi/src/parts/workspace.yaml [hunk 1]
```diff
@@ -494,6 +494,11 @@ components:
           $ref: '#/components/schemas/AzureContext'
         properties:
           $ref: '#/components/schemas/Properties'
+        lastUpdatedDate:
+          description: Lastest timestamp when the workspace is changed.
```
Comment (ddietterich) [inline openapi/src/parts/workspace.yaml:498]: Suggest: Timestamp of the latest workspace change

**Review by ddietterich — APPROVED** [review]
lgtm

**Comment by ddietterich** [comment]
> I defer to Dan on where to put this logic. My natural tendency is also to put this logic in lower level, Service, so all controller calls get the new fields automatically. Dan, just curious on why you prefer logic in controller?

Here is my take: I do not want to compute highest permission and latest update every time we build a `Workspace` object.
And I don't want to have sometimes-there-sometimes-not-there fields in the `Workspace` object. So I think fetching them should be separate. They are only consumed in the API, so making a form of get/list for the controller layer makes sense to me.

But maybe I am missing something. I'm not seeing unexpected logic in the controller.

**Comment by yuhuyoyo** [comment]
> > I defer to Dan on where to put this logic. My natural tendency is also to put this logic in lower level, Service, so all controller calls get the new fields automatically. Dan, just curious on why you prefer logic in controller?
> 
> Here is my take: I do not want to compute highest permission and latest update every time we build a `Workspace` object. And I don't want to have sometimes-there-sometimes-not-there fields in the `Workspace` object. So I think fetching them should be separate. They are only consumed in the API, so making a form of get/list for the controller layer makes sense to me.
> 
> But maybe I am missing something. I'm not seeing unexpected logic in the controller.

I was originally motivated to do this because in the controller it seems that we fetches the highest role when converting to workspace description every time. I must have missed the conversation about where this should live...

**Pull request closed without merging**